crates/re_data_store/benches/data_store.rs

```diff
@@ -86,14 +95,62 @@ fn insert(c: &mut Criterion) {
     }
 }
 
+fn insert_same_time_point(c: &mut Criterion) {
+    // Benchmark a corner-case where all rows have the same time point, and arrive out-of-order.
+    // See https://github.com/rerun-io/rerun/issues/4415
+
+    for num_rows in [1000, 10_000, 100_000] {
+        for shuffled in [false, true] {
+            let num_instances = 1;
+            let packed = false;
+            let mut group = c.benchmark_group(format!(
+                "datastore/num_rows={num_rows}/num_instances={num_instances}/shuffled={shuffled}/insert_same_time_point"
```

Comment:
Benchmark-specific params go last so one can easily run just that benchmark with all its permutations:
```suggestion
                "datastore/num_rows={num_rows}/num_instances={num_instances}/insert_same_time_point/shuffled={shuffled}"